ouroboros-consensus/src-docs/Test/Tutorial.lhs

```diff
@@ -0,0 +1,716 @@
+% Example: Implementing a Simple Protocol Using `ouroborus-consensus`
+
+Generating Documentation From This File
+=======================================
+
+This file is written in a `markdown+lhs` style understood by `pandoc`.  To
+generate a document from this file - it is as simple as:
+
+```
+ pandoc -s -f markdown+lhs Tutorial.lhs -o <output file>
+```
+
+Which will work for many of the the output types supported by `pandoc`.
+
+
+Introduction and Motivation
+===========================
+
+This example is a compilable Literate Haskell (`.lhs`) file that instantiates
+the `ConsensusProtocol` typeclass to serve as an example of some of the
+high-level concepts in `ouroboros-consensus`
+
+This example uses several extensions:
+
+> {-# OPTIONS_GHC -Wno-unused-top-binds   #-}
+> {-# LANGUAGE TypeFamilies               #-}
+> {-# LANGUAGE DerivingVia                #-}
+> {-# LANGUAGE DataKinds                  #-}
+> {-# LANGUAGE DeriveGeneric              #-}
+> {-# LANGUAGE FlexibleInstances          #-}
+> {-# LANGUAGE DeriveAnyClass             #-}
+> {-# LANGUAGE MultiParamTypeClasses      #-}
+> {-# LANGUAGE StandaloneDeriving         #-}
+
+> module Test.Tutorial () where
```

Comment:
suggestion: rename the module to something more obvious, like `Ouroboros.Consensus.Tutorial.Simple`